pytorch_lightning/utilities/types.py

```diff
@@ -43,7 +56,30 @@
     Dict[str, Sequence[DataLoader]],
 ]
 EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
-# todo: improve LRSchedulerType naming/typing
+
+
+@runtime_checkable
+class _LRScheduler(Protocol):
+    optimizer: Optimizer
+
```

Comment:
@carmocca 
I couldn't figure out what's the best way to tell mypy that LRScheduler has an attribute "optimizer". Is this the right way?

Reply:
Mmm I don't know for sure.

AFAIK the runtime checkable is for `isinstance` checks
And I don't know if it works with class attributes like that.

Ideally they would define it here:https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.pyi#L4-L10
Or you would extend that

Reply:
updated as discussed

Reply:
unfortunately, the changes I make here are messing with LightningCLI, not sure yet why

Reply:
Pushed a commit that fixes the CLI, but breaks mypy, we might need separate variables.

Feel free to revert